endgame_postprocessing/model_wrappers/lf/testRun.py

```diff
@@ -12,57 +18,147 @@
-        "year_of_threshold_prevalence_avg",
-        "year_of_90_under_threshold",
-    ],
-    threshold_groupby_cols=constants.COUNTRY_THRESHOLD_SUMMARY_GROUP_COLUMNS,
-    threshold_cols_rename=constants.COUNTRY_THRESHOLD_RENAME_MAP,
-)
-country_lvl_data.to_csv(
-    "post-processed-outputs/aggregated/combined-lf-country-lvl-agg.csv"
-)
-africa_lvl_aggregate(
-    country_lvl_data=country_lvl_data,
-    measures_to_summarize=["sampled mf prevalence (all pop)"],
-    columns_to_group_by=constants.AFRICA_LVL_GROUP_COLUMNS,
-).to_csv("post-processed-outputs/aggregated/combined-lf-africa-lvl-agg.csv")
+
+    with tqdm_joblib(total=total_ius, desc="Post-processing Scenarios"):
+
+        def single_file(file_info):
+            process_single_file(
+                raw_model_outputs=pd.read_csv(file_info.file_path),
+                scenario=file_info.scenario,
+                iuName=file_info.iu,
+                prevalence_marker_name="sampled mf prevalence (all pop)",
+                post_processing_start_time=1970,
+                measure_summary_map={
+                    "sampled mf prevalence (all pop)": measure_summary_float,
+                    "true mf prevalence (all pop)": measure_summary_float,
+                },
+            ).to_csv(
+                f"{output_iu_dir}/{file_info.scenario}_{file_info.iu}_post_processed.csv"
+            )
+
+        Parallel(num_jobs)(delayed(single_file)(file_info) for file_info in all_files)
+
+
+def generate_aggregates_for_all_ius(processed_iu_dir: str, output_aggregate_dir: str):
+    """
+    Generates the aggregate files - one for all IUs, one for all countries, one for Africa.
+    Expected to be run after run_aggregate_runs_for_each_iu
+
+    processed_iu_dir must have the following substructure:
+    processed_iu_dir/
+        a csv per IU with name format
+        scenario1_iu1_post_processed.csv
+        The output directory must be empty.
+    On completion the sub-structure will be:
+    output_aggregate_dir/
+        combined-lf-iu-lvl-agg.csv - all IUs in one csv
+            a aggregated by country csv
+        combined-lf-country-lvl-agg.csv - aggregate by country
+        combined-lf-africa-lvl-agg.csv - aggregated across Africa
+    """
+    os.makedirs(output_aggregate_dir)
+    combined_ius = aggregate_post_processed_files(processed_iu_dir)
+    aggregated_df = iu_lvl_aggregate(combined_ius)
+    aggregated_df.to_csv(f"{output_aggregate_dir}/combined-lf-iu-lvl-agg.csv")
+    country_lvl_data = country_lvl_aggregate(
+        iu_lvl_data=aggregated_df,
+        general_summary_measure_names=[
+            "sampled mf prevalence (all pop)",
+            "year_of_threshold_prevalence_avg",
+            "year_of_90_under_threshold",
+        ],
+        general_groupby_cols=constants.COUNTRY_SUMMARY_GROUP_COLUMNS,
+        threshold_summary_measure_names=[
+            "year_of_threshold_prevalence_avg",
+            "year_of_90_under_threshold",
```

Comment:
Some of these would have to be changed after this PR: https://github.com/NTD-Modelling-Consortium/endgame-postprocessing/pull/8, but I can edit that after this change goes through